src/ideviceinstaller.c

```diff
@@ -347,27 +347,43 @@ static void idevice_wait_for_command_to_complete()
 
 static int str_is_udid(const char* str)
 {
-	const char allowed[] = "0123456789abcdefABCDEF";
-
+	
 	/* handle NULL case */
 	if (str == NULL)
 		return -1;
 
 	int length = strlen(str);
 
 	/* verify length */
-	if (length != 40)
-		return -1;
+	if (length == 40) {
```

Comment:
this code has quite a few magic numbers, which are good candidates for being moved into constants